File: Modules/SEDO/EMS/SIE.Web.EMS/Tpms/Commands/AddRecordCommand.cs
using SIE.EMS.Tpms;
using SIE.Web.Command;

namespace SIE.Web.EMS.Tpms.Commands
{
    /// <summary>
    /// TPM添加命令
    /// </summary>
    [JsCommand("SIE.Web.EMS.Tpms.Commands.AddRecordCommand")]
    public class AddRecordCommand : ViewCommand
    {
        /// <summary>
        /// 添加命令
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>true</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var data = args.Data.ToJsonObject<TpmRecord>();
            data.TpmNo = RT.Service.Resolve<TpmController>().GetTpmScoreNo();
            return data;
        }
    }
}

File: Modules/SEDO/EMS/SIE.Web.EMS/Tpms/Commands/AddRecordCommand.js
SIE.defineCommand('SIE.Web.EMS.Tpms.Commands.AddRecordCommand', {
    extend: 'SIE.cmd.Add',
    meta: { text: "添加", group: "edit", iconCls: "iconfont icon-AddEntity icon-green" },
     execute: function (view, source) {
         var editEntity = this.getEditEntity();
         CRT.Workbench.addPage({
            tabId: 'tab_addRecord',
            title: '添加-TPM操作记录',
            entityType: view.model,
            recordId: editEntity.data.Id,
            entityType: 'SIE.EMS.Tpms.TpmRecord', 
            viewGroup:"TpmRecordAddView",
            isDetail:true,
        });
    },
});